src/quick-lint-js/fe/parse-statement.cpp

```diff
@@ -3977,6 +3977,11 @@ void Parser::parse_and_visit_if(Parse_Visitor_Base &v) {
     parse_and_visit_body();
     break;
 
+  case Token_Type::semicolon:
```

Comment:
Nitpick: It's helpful to include a comment explaining what code this code path covers. For example:
```suggestion
  // if (cond);  // Invalid TypeScript.
  case Token_Type::semicolon:
```